src/internal/App.ts

```diff
@@ -155,6 +157,47 @@ export default class App extends EventHandler implements IView<App>, IVisStateAp
     return spec;
   }
 
+  /**
+   * Switch the currently loaded Vega specification.
+   * Push the new specification to the provenance graph and open it in the vega view
+   * @param {Spec} spec
+   * @param {string} title
+   */
+  private switchVegaSpec(spec: Spec, title: string) {
+    const bak = this.currentSpec;
+    this.currentSpec = spec;
+    this.graph
```

Comment:
if you execute the function anyhow, there is no need for "pushWithResult" but just a regular "push"